internal/pkg/githubapi/github.go

```diff
@@ -230,6 +226,54 @@ func HandlePREvent(eventPayload *github.PullRequestEvent, ghPrClientDetails GhPr
 	}
 }
 
+func generateArgoCdDiffComments(diffCommentData DiffCommentData, githubCommentMaxSize int) (comments []string, err error) {
+	err, templateOutput := executeTemplate("argoCdDiff", "argoCD-diff-pr-comment.gotmpl", diffCommentData)
+	if err != nil {
+		return comments, fmt.Errorf("Failed to generate ArgoCD diff comment template: err=%s\n", err)
+	}
+
+	// Happy path, the diff comment is small enough to be posted in one comment
+	if len(templateOutput) < githubCommentMaxSize {
+		comments = append(comments, templateOutput)
+		return comments, err
+	}
+
+	// If the diff comment is too large, we'll split it into multiple comments, one per component
+	totalComponents := len(diffCommentData.DiffOfChangedComponents)
+	for i, singleComponentDiff := range diffCommentData.DiffOfChangedComponents {
+		// We take the diffCommentData and replace the DiffOfChangedComponents with a single component diff
+		componentTemplateData := diffCommentData
+		componentTemplateData.DiffOfChangedComponents = []argocd.DiffResult{singleComponentDiff}
+		// We also update the header to reflect the current component.
+		componentTemplateData.Header = fmt.Sprintf("Component %d/%d: %s(Splited for comment size)", i+1, totalComponents, singleComponentDiff.ComponentPath)
+		err, templateOutput := executeTemplate("argoCdDiff", "argoCD-diff-pr-comment.gotmpl", componentTemplateData)
+		if err != nil {
+			return comments, fmt.Errorf("Failed to generate ArgoCD diff comment template: err=%s\n", err)
+		}
+		// Even per component comments can be too large, in that case we'll just use the concise template
+		// Somewhat Happy path, the per-component diff comment is small enough to be posted in one comment
+		if len(templateOutput) < githubCommentMaxSize {
+			comments = append(comments, templateOutput)
+			if err != nil {
+				return comments, err
+			}
```

Comment:
Which error are we checking here?

Reply:
Nice catch, its a leftover from before a refactoring I did (at line 256 I used to actually post the comment to GH API)
I later changed to building an array of comments and posting them to GH API outside of the function to make it more testable.

Reply:
I'll remove it